RGIS2_MergingSpatialData_part1_Joins.Rmd

```diff
@@ -70,24 +70,13 @@ where:
 * _"id-number"_ is the colum that contains the unique identifier in _worldCountries_, and 
 * _"countryID"_ is the column that contains the unique identifier in _countryData_. 
 
-Then we can just merge worldCountries@data with countryData on those variables, with one small nuance: 
-
-**DO NOT MERGE YOUR DATA USING THE `merge` COMMAND!** 
-
-R keeps track of the relationship between rows in the attribute table an `Spatial*` shapes based on the order of observations in the attribute table. Unfortunately, the `merge` command can shuffle the order of rows in the attribute table, potentially jumbling the association of rows with Point/Line/Polygons.
-
-To avoid this problem, always use two rules when merging:
-* Use `join` from the `plyr` library with the `type="left"` argument
-* Also place the DataFrame associated with your `Spatial*` object in the first position. 
+Then we can just merge worldCountries with countryData on those variables, using the `merge` method in `sp`:
```

Comment:
Answer to your question - Yes. In this case R is not masking functions, it is extending them using the generic function abilities of R. This is nothing to be afraid of. Just be aware that when the first argument is a Spatial*DataFrame the behavior is customized. As well as preserving order it also copies the coordinate data into the result and returns a Spatial* object.